benchalerts/benchalerts/conbench_dataclasses.py

```diff
@@ -190,14 +186,34 @@ class FullComparisonInfo:
     the comparisons to their baselines.
     """
 
+    # Can be empty.
     run_comparisons: List[RunComparisonInfo]
 
-    def __post_init__(self):
-        # Any code that constructs an instance should ensure run_comparisons is non-empty
-        assert self.run_comparisons
+    @property
+    def has_any_contender_runs(self) -> bool:
+        """Whether there are any contender runs available."""
+        return bool(self.run_comparisons)
+
+    @property
+    def has_any_contender_results(self) -> bool:
+        """Whether there are any contender benchmark results available."""
+        for run_comparison in self.run_comparisons:
+            if run_comparison.contender_benchmark_result_info:
+                return True
+        return False
+
+    @property
+    def has_any_z_analyses(self) -> bool:
+        """Whether there are any lookback z-score analyses available."""
+        for run_comparison in self.run_comparisons:
+            if run_comparison.compare_results:
+                for compare_result in run_comparison.compare_results:
+                    if compare_result["analysis"]["lookback_z_score"]:
+                        return True
+        return False
 
     @property
-    def benchmarks_with_errors(self) -> List[BenchmarkResultInfo]:
+    def results_with_errors(self) -> List[BenchmarkResultInfo]:
```

Comment:
Another clarifying rename (it doesn't really have to do with this PR).

Reply:
Feels good. :)